### ISSUE TITLE
Implement classes from sklearn.neighbors: Nearest Neighbors

### ISSUE THREAD

**Issue by arpieb** [arpieb/nixa#8]
Implement some or all (as makes sense) from `sklearn.nieghbors`:

- [ ] [BallTree](https://scikit-learn.org/stable/modules/generated/sklearn.neighbors.BallTree.html#sklearn.neighbors.BallTree)(X[, leaf_size, metric]) - BallTree for fast generalized N-point problems
- [x] [KDTree](https://scikit-learn.org/stable/modules/generated/sklearn.neighbors.KDTree.html#sklearn.neighbors.KDTree)(X[, leaf_size, metric]) - KDTree for fast generalized N-point problems
- [ ] [KernelDensity](https://scikit-learn.org/stable/modules/generated/sklearn.neighbors.KernelDensity.html#sklearn.neighbors.KernelDensity)(*[, bandwidth, ...]) - Kernel Density Estimation.
- [ ] [KNeighborsClassifier](https://scikit-learn.org/stable/modules/generated/sklearn.neighbors.KNeighborsClassifier.html#sklearn.neighbors.KNeighborsClassifier)([...]) - Classifier implementing the k-nearest neighbors vote.
- [ ] [KNeighborsRegressor](https://scikit-learn.org/stable/modules/generated/sklearn.neighbors.KNeighborsRegressor.html#sklearn.neighbors.KNeighborsRegressor)([n_neighbors, ...]) - Regression based on k-nearest neighbors.
- [ ] [KNeighborsTransformer](https://scikit-learn.org/stable/modules/generated/sklearn.neighbors.KNeighborsTransformer.html#sklearn.neighbors.KNeighborsTransformer)(*[, mode, ...]) - Transform X into a (weighted) graph of k nearest neighbors.
- [ ] [LocalOutlierFactor](https://scikit-learn.org/stable/modules/generated/sklearn.neighbors.LocalOutlierFactor.html#sklearn.neighbors.LocalOutlierFactor)([n_neighbors, ...]) - Unsupervised Outlier Detection using the Local Outlier Factor (LOF).
- [ ] [RadiusNeighborsClassifier](https://scikit-learn.org/stable/modules/generated/sklearn.neighbors.RadiusNeighborsClassifier.html#sklearn.neighbors.RadiusNeighborsClassifier)([...]) - Classifier implementing a vote among neighbors within a given radius.
- [ ] [RadiusNeighborsRegressor](https://scikit-learn.org/stable/modules/generated/sklearn.neighbors.RadiusNeighborsRegressor.html#sklearn.neighbors.RadiusNeighborsRegressor)([radius, ...]) - Regression based on neighbors within a fixed radius.
- [ ] [RadiusNeighborsTransformer](https://scikit-learn.org/stable/modules/generated/sklearn.neighbors.RadiusNeighborsTransformer.html#sklearn.neighbors.RadiusNeighborsTransformer)(*[, ...]) - Transform X into a (weighted) graph of neighbors nearer than a radius.
- [ ] [NearestCentroid](https://scikit-learn.org/stable/modules/generated/sklearn.neighbors.NearestCentroid.html#sklearn.neighbors.NearestCentroid)([metric, ...]) - Nearest centroid classifier.
- [ ] [NearestNeighbors](https://scikit-learn.org/stable/modules/generated/sklearn.neighbors.NearestNeighbors.html#sklearn.neighbors.NearestNeighbors)(*[, n_neighbors, ...]) - Unsupervised learner for implementing neighbor searches.
- [ ] [NeighborhoodComponentsAnalysis](https://scikit-learn.org/stable/modules/generated/sklearn.neighbors.NeighborhoodComponentsAnalysis.html#sklearn.neighbors.NeighborhoodComponentsAnalysis)([...]) - Neighborhood Components Analysis.
